### PR TITLE
Remove InfiniteSampler from base configs for COCO

### PR DESCRIPTION
**Pull request by BloodAxe**
This PR removes `InfiniteSampler` from default COCO dataset config. 

The motivation for this change is this class causes random crashes in DataLoader with num_workers > 0. 
It was not traced till the very end, but presumably due to the way this class is implemented (Sampler has no "ending"). 

For instance, following snippet cause crash on Windows:

```

if __name__ == "__main__":
    dataset = list(range(1123123))

    sampler = InfiniteSampler(dataset, shuffle=True, seed=0, rank=0, world_size=1)

    for batch in tqdm(DataLoader(dataset, batch_size=32, drop_last=False, sampler=sampler, num_workers=4), desc="InfiniteSampler"):
        pass

    print("Iterationg over dataset done")

# InfiniteSampler: 100%|█████████▉| 35060/35098 [00:37<00:00, 1220.31it/s]
# Process finished with exit code -1073741819 (0xC0000005)

```

It also ensures that `drop_last: True` is set to avoid rare but possible case when last batch would have size 1 and this would crash training of the model with BatchNormalization layers (They cannot operate on bs=1 in train).

`InfiniteSampler` class is kept in place to avoid breaking change, but annotated with `@deprecated` and will be removed in 3.1.0.

### CODE REVIEW

**Review by shaydeci — APPROVED** [review]
LGTM

**Review by Louis-Dupont — APPROVED** [review]
LGTM, nice catch
